normandy/control/static/control/js/components/RecipeForm.jsx

```diff
@@ -1,34 +1,90 @@
 import React from 'react'
 import { connect } from 'react-redux'
 import { Link } from 'react-router'
+import { destroy, reduxForm, getValues } from 'redux-form'
+import { _ } from 'underscore'
+
+import apiFetch from '../utils/apiFetch.js';
+import { reduxFormFields } from '../utils/formHelpers.js';
+
 import ControlActions from '../actions/ControlActions.js'
-import { reduxForm } from 'redux-form'
 import composeRecipeContainer from './RecipeContainer.jsx'
+import ActionForm from './ActionForm.jsx'
 
-
-class RecipeForm extends React.Component {
+export class RecipeForm extends React.Component {
   constructor(props) {
     super(props);
     this.submitForm = this.submitForm.bind(this);
+    this.getAvailableActions = this.getAvailableActions.bind(this);
+    this.changeAction = this.changeAction.bind(this);
+
+    this.state = {
+      availableActions: [],
+      selectedAction: null,
+    };
+  }
+
+  getAvailableActions(recipeId) {
+    apiFetch('/api/v1/action/')
+    .then(availableActions => {
+      let selectedActionName = (this.props.recipe ? this.props.recipe.action_name : null);
+
+      availableActions.map(action => {
+        action.fields = reduxFormFields[action.name];
+
+        if (selectedActionName === action.name) {
+          this.setState({
+            selectedAction: action
+          });
+        }
+
+        return action;
+      });
```

Comment:
`Array.map` is generally used for making a new list that has be transformed, not for modifying a list in place. In fact, this is sort of throwing away a lot of things, since it is making an entirely new list and then not savings it. This would probably be better as `availableActions.forEach`, which doesn't build the new list, or as a `for (let action in availableActions) { ... }`.